### DISCUSSION TITLE
JSON 编码mega-layout布局问题 自定义组件

### DISCUSSION THREAD

**Discussion by LXY-nice** [alibaba/formily#1387] [Q&A 1.x]
用mega-layout 布局自定义组件  没办法宽度撑满 设置了 "full": true也无效  
![image](https://user-images.githubusercontent.com/59158375/117768609-2e1c9c00-b265-11eb-83f1-599b648c5b93.png)

**Comment by LXY-nice** [comment]
解决办法在定义组件内要设置下width 100% 因为felx了  
其他疑问
![image](https://user-images.githubusercontent.com/59158375/117773880-717a0900-b26b-11eb-9076-c2a7164cb496.png)
动态变化组件  字段不变   有办法可以动态替换组件吗

**Comment by shjames** [comment]
文档哪里有说 mega-layout这个组件的用法的？怎么找不到
